### DISCUSSION TITLE
SPX Snapshot & SplitStoreV2 - Testing & Call Outs

### DISCUSSION THREAD

**Discussion by TippyFlitsUK** [filecoin-project/lotus#9179] [Show and tell]
![SPX_Snap_Split_Banner](https://user-images.githubusercontent.com/3090542/184678118-f308d41c-3890-41f3-bdb9-22042c5fae34.png)

## Hello SPX Fellows

&nbsp;
Thank you to everyone who participated in the [initial SpiltStoreV2 testing](https://github.com/filecoin-project/lotus/discussions/8927) round!!

@zenground0 has been working tirelessly on the upcoming SplitStoreV2 rollout and we have a shiny new release for you to break :wink:. 

Thanks to @travisperson we also have the brand new Snapshot Service which was [soft-launched](https://filecoinproject.slack.com/archives/CPFTWMY7N/p1659627726732259)  on 4th August 2022! You can read more about the new service on [this public Notion page](https://pl-strflt.notion.site/pl-strflt/Lightweight-Filecoin-Chain-Snapshots-17e4c386f35c44548f5863afb7b5e024).

It goes without saying that both of these new releases represent a huge improvement over the current implementations!

We would sincerely appreciate your help in testing both of these new enhancements and seeing as enabling splitstore ideally also involves a new snapshot import, we're going to test both at the same time. This assignment is open to all SPX fellows, please see the details below to get started,

This assignment is applicable to both mainnet and calibration instances but we are particularly interested in hearing about your mainnet experiences. In the event of any critical failures on your node, recovery is a very familiar procedure as detailed below.

Please reach out to @jennijuju or @TippyFlits on Slack if you have any questions! :pray:
</br>
## Step 1: ⚠️ Backup ALL lotus wallets before doing anything! (`lotus wallet list`) ⚠️
```
lotus wallet export <wallet-address-1> > file-1
lotus wallet export <wallet-address-2> > file-2
lotus wallet export <wallet-address-3> > file-3
etc...
```

## Step 2: Download a new lightweight snapshot & Update to testing release

- You are all well used to downloading snapshots and have your own techniques already in place. Please note the new snapshot links as listed on the [Notion page](https://pl-strflt.notion.site/pl-strflt/Lightweight-Filecoin-Chain-Snapshots-17e4c386f35c44548f5863afb7b5e024) and [Slack announcement](https://filecoinproject.slack.com/archives/CPFTWMY7N/p1659627726732259): 
Mainnet - https://snapshots.mainnet.filops.net/minimal/latest
Calibration - https://snapshots.calibrationnet.filops.net/minimal/latest
- Upgrade both daemon and miner instances to: [1.17.1-rc3](https://github.com/filecoin-project/lotus/releases/tag/v1.17.1-rc3)

## Step 3: Enable SplitStore and configure ColdStore mode
SplitStore can be run in `universal`,`discard` and now `AutoPrune` modes and all three modes have been enhanced since the previous testing round. We are looking for a selection of SPX fellows to test a combination of three. Select 🎉  if you are testing `universal` mode and 🚀  if you are opting for `discard` mode and 👀 if your are testing `AutoPrune`. Please review the emoji count at the bottom of this post before selecting a mode, an even balance of the 3 options would be much appreciated. The selected mode can be set in your `/.lotus/config.toml` as follows:
| Universal Mode | Discard Mode | AutoPrune Mode |
| --- | --- | --- |
| [Chainstore]</br>&nbsp;&nbsp;EnableSplitstore = true</br>&nbsp;&nbsp;[Chainstore.Splitstore]</br>&nbsp;&nbsp;&nbsp;&nbsp;ColdStoreType = "universal"</br>&nbsp;&nbsp;&nbsp;&nbsp;EnableColdStoreAutoPrune = false | [Chainstore]</br>&nbsp;&nbsp;EnableSplitstore = true</br>&nbsp;&nbsp;[Chainstore.Splitstore]</br>&nbsp;&nbsp;&nbsp;&nbsp;ColdStoreType = "discard"</br>&nbsp;&nbsp;&nbsp;&nbsp;EnableColdStoreAutoPrune = false | [Chainstore]</br>&nbsp;&nbsp;EnableSplitstore = true</br>&nbsp;&nbsp;[Chainstore.Splitstore]</br>&nbsp;&nbsp;&nbsp;&nbsp;ColdStoreType = "universal"</br>&nbsp;&nbsp;&nbsp;&nbsp;EnableColdStoreAutoPrune = true |

## Step 4: Prepare your `/.lotus/datastore` and import snapshot
- Stop all miner and daemon instances
- If you are **not** yet running Splitstore - Delete all files in your `/.lotus/datastore/chain` folder.
- If you are **already running** SplitStore - Delete all files in your `/.lotus/datastore/chain` folder and clear your `/.lotus/datastore/splitstore` folder by running `./lotus-shed splitstore clear --repo=/path/to/your/lotus/repo`
- Import your new lightweight snapshot in the normal manner
- Please make a note of your `/.lotus/datastore/chain` and `/.lotus/datastore/splitstore` folder sizes immediately following snapshot import and successful chain sync completion and record your results in the reply card below.
- Set `lotus log set-level --system splitstore debug`

## Step 5: Reply card
Please provide the information below after completion of the above steps and immediately following the successful start-up of your SplitStore enabled daemon:

- [ ] - If the new lightweight snapshot downloads correctly
- [ ] - If the new lightweight snapshot imports correctly

| SplitStore Mode | Directory | Size in GiBs immediately following daemon start |
| --- | --- | --- |
|  | /.lotus/datastore/chain |  |
|  | /.lotus/datastore/splitstore |  |

</br>Please complete the following after observing usual operations:
- [ ] - If `lotus daemon` stays in sync
- [ ] - if you can see the following new symlink path - `/.lotus/datastore/splitstore/hot.badger` 🔥 🦡 
- [ ] - if your `/.lotus/datastore/chain` folder is being successfully garbage collected
- [ ] - if normal mining tasks appear to be processed as expected
- [ ] - if normal daemon tasks appear to be processed as expected

We will be asking for updated sizes of your `/.lotus/datastore/chain` and `/.lotus/datastore/splitstore` folders on a weekly basis for a period of 4 weeks. We will publish weekly reminders in `#spx-current-class` but this is obviously dependent on your testing start date. Sizes should be recorded at the end of each week and in `GiB` format. Please consider setting weekly reminders so metrics remain consistent and accurate. :pray:
 
| SplitStore Mode | Directory | Week 1 | Week 2 | Week 3 | Week 4 |
| --- | --- | --- | --- | --- | --- |
|  | /.lotus/datastore/splitstore |  |  |  |  |
|  | /.lotus/datastore/chain |  |  |  |  |

## Reminders and Risk Mitigations
In the event of a critical failure or data corruption, you will need to re-import a lightweight snapshot with the following steps:

- ❗ Ensure that all lotus wallets are fully backed up.
- Stop lotus and delete your `/.lotus /datastore` folder 
- Download a new lightweight snapshot
- Disable SplitStoreV2 in your `/.lotus/config.toml` - `EnableSplitstore = false`

📓 In the event that you are unable to start your daemon after trying the above steps, please delete just the folders in your `/.lotus` directory and retry the lightweight snapshot import. ❗ ONLY DELETE THE FOLDERS - the `api`, `token` and `config.toml` files should not be deleted!

Questions are welcomed in this discussion, the team will be monitoring and will check in regularly.

Issues can be reported at https://github.com/filecoin-project/lotus/issues. Issues should be submitted with:

- Daemon logs 
- GO routine dumps `lotus pprof goroutines` and `lotus-miner pprof goroutines`
- The operation/step at which the problem occurred
- The stack trace or logs as relevant. As always, for large log files, you may upload them to this [Google Drive folder](https://drive.google.com/drive/folders/1MKHhsNVRAP8HOeBmd1hq9lethLOKU5xl) and link them in the issue.

📓 And thank you to Reiers Productions for the very cool banner!!

**Comment by stuberman** [comment]
- [x]  - If the new lightweight snapshot downloads correctly
- [x]  - If the new lightweight snapshot imports correctly

SplitStore Mode | Directory | Size in GiBs immediately following daemon start
-- | -- | --
AutoPrune  | /.lotus/datastore/chain |  84G on start - 84G after sync
AutoPrune  | /.lotus/datastore/splitstore |  1.2G on start - 23G after sync




SplitStore Mode | Directory | Week 1 | Week 2 | Week 3 | Week 4 | Week 5
-- | -- | -- | -- | -- | --  | --
AutoPrune | /.lotus/datastore/splitstore | 159G | 163G | 153G |  161G |  178G
AutoPrune  | /.lotus/datastore/chain | 244G | 403G | 631G |  831G |  1.1T

- [x]  - If lotus daemon stays in sync
- [x]  - if you can see the following new symlink path - /.lotus/datastore/splitstore/hot.badger 🔥 🦡
- [ ]  - if your /.lotus/datastore/chain folder is being successfully garbage collected
- [x]  - if normal mining tasks appear to be processed as expected
- [x]  - if normal daemon tasks appear to be processed as expected

**Reply by stuberman** [reply]
Interesting log entries

> 2022-08-18T23:07:49.196 INFO bellperson::gpu::multiexp > Please use `RUST_GPU_TOOLS_CUSTOM_GPU` instead of `BELLMAN_CUSTOM_GPU`, their values are fully compatible.
> 2022-08-18T23:07:49.196 INFO rust_gpu_tools::corecounts > Adding "NVIDIA GeForce RTX 2080 Ti" to GPU list with 4352 CUDA cores.

**Reply by TippyFlitsUK** [reply]
Well, that's new!! Thanks Stu, I will liaise with the team and get back to you.

**Reply by TippyFlitsUK** [reply]
Just to confirm Stu, this is as a result of the new proofs release and nothing to be concerned about. Interesting spot, we don't often see new random new log entries in the wild these days!! :wink:

**Reply by stuberman** [reply]
I see the new path: `/.lotus/datastore/splitstore/hot.badger`
But is a symlink supposed to be created already? Mine is not symlinked.
** Update *** Now there is a symlink as of Aug 20

**Reply by stuberman** [reply]
> Just to confirm Stu, this is as a result of the new proofs release and nothing to be concerned about. Interesting spot, we don't often see new random new log entries in the wild these days!! 😉

I replaced my Bellman var with the RUST var, I am assuming that is the new var? Or should I return it to the Bellman var?

**Reply by TippyFlitsUK** [reply]
No need to change back Stu, this will be the default going forward.

**Comment by tmyuu** [comment]
## Step 4:
```
$ rm -rf /usr/local/filecoin/.node/datastore/chain/*
$ ./lotus-shed splitstore clear --repo=/usr/local/filecoin/.node
clearing splitstore directory...
deleting splitstore keys from metadata datastore...
2022-08-19T14:07:29.265+0900	INFO	badger	v2@v2.2007.3/levels.go:183	All 0 tables opened in 0s

2022-08-19T14:07:29.268+0900	INFO	badger	v2@v2.2007.3/value.go:1158	Replaying file id: 0 at offset: 0

2022-08-19T14:07:29.268+0900	INFO	badger	v2@v2.2007.3/value.go:1178	Replay took: 4.028µs

2022-08-19T14:07:29.349+0900	INFO	badger	v2@v2.2007.3/levels.go:183	All 0 tables opened in 0s

2022-08-19T14:07:29.351+0900	INFO	badger	v2@v2.2007.3/value.go:1158	Replaying file id: 0 at offset: 0

2022-08-19T14:07:29.352+0900	INFO	badger	v2@v2.2007.3/value.go:1178	Replay took: 4.789µs

deleting /splitstore/baseEpoch from datastore...
deleting /splitstore/compactionIndex from datastore...
deleting /splitstore/markSetSize from datastore...
deleting /splitstore/pruneEpoch from datastore...
deleting /splitstore/warmupEpoch from datastore...
```
## Step5:
* [x]  - If the new lightweight snapshot downloads correctly
* [x]  - If the new lightweight snapshot imports correctly

| SplitStore Mode | Directory | Size in GiBs immediately following daemon start |
| - | - | - |
| AutoPrune Mode | /.lotus/datastore/chain | 83G |
| AutoPrune Mode | /.lotus/datastore/splitstore | 85G |

Please complete the following after observing usual operations:
* [x]  - If `lotus daemon` stays in sync
* [x]  - if you can see the following new symlink path - `/.lotus/datastore/splitstore/hot.badger` 🔥 🦡
* [x]  - if your `/.lotus/datastore/chain` folder is being successfully garbage collected
* [x]  - if normal mining tasks appear to be processed as expected
* [x]  - if normal daemon tasks appear to be processed as expected

Sizes of the folders on a weekly basis for a period of 4 weeks:
| SplitStore Mode | Directory | Week 1 | Week 2 | Week 3 | Week 4
| - | - | - | - | - | - |
| AutoPrune Mode | /.lotus/datastore/splitstore | 144GB | 321GB | | |
| AutoPrune Mode | /.lotus/datastore/chain | 243GB | 436GB | | |

| SplitStore Mode | Directory | Week 1 | Week 2 | Week 3 | Week 4
| - | - | - | - | - | - |
| Discard Mode | /.lotus/datastore/splitstore | | | | |
| Discard Mode | /.lotus/datastore/chain | | | | |

**Comment by benjaminh83** [comment]
- [x]  If the new lightweight snapshot downloads correctly
- [x]  If the new lightweight snapshot imports correctly

SplitStore Mode | Directory | Size in GiBs immediately following daemon start
-- | -- | --
AutoPrune | /.lotus/datastore/chain | 85G
AutoPrune | /.lotus/datastore/splitstore | 9G

- [x]  If lotus daemon stays in sync
- [x]  if you can see the following new symlink path - /.lotus/datastore/splitstore/hot.badger 🔥 🦡
- [ ]  if your /.lotus/datastore/chain folder is being successfully garbage collected
- [x]  if normal mining tasks appear to be processed as expected
- [x]  if normal daemon tasks appear to be processed as expected
 
Sizes of the folders on a weekly basis for a period of 4 weeks:

SplitStore Mode | Directory | Week 1 | Week 2 | Week 3 | Week 4
-- | -- | -- | -- | -- | --
AutoPrune Mode | /.lotus/datastore/splitstore | 86G | 97G | 90G | 104G 
AutoPrune Mode | /.lotus/datastore/chain | 238G | 442G | 646G | 793G
 
 
 ### Round2 on v1.17.2-rc2
 SplitStore Mode | Directory | Week 1 | Week 2 | Week 3 | Week 4
-- | -- | -- | -- | -- | --
AutoPrune Mode | /.lotus/datastore/splitstore | 173G | | | 
AutoPrune Mode | /.lotus/datastore/chain | 89G | | |

**Reply by benjaminh83** [reply]
@TippyFlitsUK I'm running: 
```
export LOTUS_CHAINSTORE_ENABLESPLITSTORE=true
export LOTUS_CHAINSTORE_SPLITSTORE_COLDSTORETYPE="universal"
export LOTUS_CHAINSTORE_SPLITSTORE_ENABLECOLDSTOREAUTOPRUNE=true
```
Why does my chain folder keep growing? It was never this big when I just ran coldstoretype discard...

**Reply by TippyFlitsUK** [reply]
Hey Ben :wave:
The ColdStore GC happens about once a week so it may not have triggered just yet. Definitely worth monitoring though!! Let me know if you're still seeing this when we get to the end of this week :pray:
```
    # ColdStoreFullGCFrequency specifies how often to performa a full (moving) GC on the coldstore.
    # Only applies if auto prune is enabled.  A value of 0 disables while a value of 1 will do
    # full GC in every prune.
    # Default is 7 (about once every a week)
    #
    # type: uint64
    # env var: LOTUS_CHAINSTORE_SPLITSTORE_COLDSTOREFULLGCFREQUENCY
    #ColdStoreFullGCFrequency = 7
```

**Reply by TippyFlitsUK** [reply]
Ran a few tests overnight @benjaminh83 and I think we may actually have a minor issue here. Nothing show-stopping but it does indeed appear that auto-pruning is not getting triggered as it should. Zen is OOO for a few days but will raise it with him ASAP!

I'm running the same config as you and going to keep my config unchanged for the time being to monitor this over as long a period as possible. Would you be able to do the same? I appreciate this is using a lot of disk space so completely understand if it's not a viable option for you!!

**Reply by benjaminh83** [reply]
Yeah, I agree. No pruning going on here as well. I'm not at: 
```
$ du .lotus/datastore/splitstore -h
90G     .lotus/datastore/splitstore
$ du .lotus/datastore/chain -h
274G    .lotus/datastore/chain
```

**Reply by benjaminh83** [reply]
@TippyFlitsUK this is bad. Lotus lost sync and wont sync up
![image](https://user-images.githubusercontent.com/14029124/188330133-750c6aca-9c87-4e6f-8d17-f6903349a89f.png)

```
$ du .lotus/datastore/splitstore -h
92G     .lotus/datastore/splitstore

$ du .lotus/datastore/chain/ -h
416G    .lotus/datastore/chain/
```

**Reply by benjaminh83** [reply]
I had to kill the lotus daemon process. Now spinning it up again, and it was way off - like 4h behind: 

![image](https://user-images.githubusercontent.com/14029124/188330389-5c9d1598-ee2a-4e3a-8e1c-4e22dfd932a3.png)


This is not stable 👎

**Reply by TippyFlitsUK** [reply]
I'm not sure this is SplitStore related Ben. I'm not seeing any sync issues at all with the following:
```
Sun 4 Sep 19:47:52 BST 2022
162G	/filecoin-sealing/.lotus-daemon/datastore/splitstore
749G	/filecoin-storage/chain
```

**Reply by TippyFlitsUK** [reply]
Have you got enough space left in your `/.lotus/datastore/chain/` folder?

**Reply by benjaminh83** [reply]
55% used, 752GB left... This is not the problem. This was just lotus lost sync. Server htop shows nothing unusual. It's syncing up now.

**Reply by TippyFlitsUK** [reply]
Hey Ben :wave:. How are things looking on your node?

**Reply by benjaminh83** [reply]
Hi @TippyFlitsUK seems stable for now. I just don't like that my daemon randomly lost sync. I don't think that has happened before in 2022. Lets hope there is a fix to the autoprune soon, so this can get stable.

**Reply by TippyFlitsUK** [reply]
I'm afraid the fix most likely won't make it into 1.17.1 Ben. I would suggest switching to discard when you next have a chance. I will stick with the current AutoPrune config and hope it will result in a repro in the next few days.

**Reply by benjaminh83** [reply]
@TippyFlitsUK tried to do a normal withdraw from the miner. It did succeed, but command in it self did not return the "waiting for 6 epochs", and when terminating the command, it threw up like: 

```
$ lotus-miner actor withdraw 400
^Cpanic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x2134fc5]

goroutine 1 [running]:
main.glob..func2(0xc001124c00)
        /home/test/v1.17.1-rc3/cmd/lotus-miner/actor.go:255 +0x665
github.com/urfave/cli/v2.(*Command).Run(0x75b0c40, 0xc001124b00)
        /home/test/go/pkg/mod/github.com/urfave/cli/v2@v2.8.1/command.go:169 +0x6be
github.com/urfave/cli/v2.(*App).RunAsSubcommand(0xc000683380, 0xc001124940)
        /home/test/go/pkg/mod/github.com/urfave/cli/v2@v2.8.1/app.go:490 +0xa26
github.com/urfave/cli/v2.(*Command).startApp(0x75af800, 0xc001124940)
        /home/test/go/pkg/mod/github.com/urfave/cli/v2@v2.8.1/command.go:287 +0x77b
github.com/urfave/cli/v2.(*Command).Run(0xc001124940, 0x7ffecb228147)
        /home/test/go/pkg/mod/github.com/urfave/cli/v2@v2.8.1/command.go:95 +0x82e
github.com/urfave/cli/v2.(*App).RunContext(0xc000683040, {0x4465430, 0xc000136870}, {0xc000132000, 0x4, 0x4})
        /home/test/go/pkg/mod/github.com/urfave/cli/v2@v2.8.1/app.go:341 +0x89c
github.com/urfave/cli/v2.(*App).Run(...)
        /home/test/go/pkg/mod/github.com/urfave/cli/v2@v2.8.1/app.go:247
github.com/filecoin-project/lotus/cli.RunApp(0x392b020)
        /home/test/v1.17.1-rc3/cli/helper.go:35 +0x4e
main.main()
        /home/test/v1.17.1-rc3/cmd/lotus-miner/main.go:168 +0x10f2
```

Could it be that something has broken in the code that represents the output? Seems like daemon did its job and funds was transferred correctly.

**Reply by benjaminh83** [reply]
normal lotus send messages goes without issues

**Reply by TippyFlitsUK** [reply]
Thanks Ben :pray:

This is a [known issue](https://github.com/filecoin-project/lotus/issues/9154).  It has been resolved but it is not in v.1.17.1-rc3.

**Reply by benjaminh83** [reply]
@TippyFlitsUK We should take another round on this using v.1.17.2.-rc2. So I have reverted over again to:
```
export LOTUS_CHAINSTORE_SPLITSTORE_COLDSTORETYPE="universal"
export LOTUS_CHAINSTORE_SPLITSTORE_ENABLECOLDSTOREAUTOPRUNE=true
```
Tracking it once again in here :))))

**Reply by TippyFlitsUK** [reply]
Thanks Ben!! Really appreciate your efforts but I'm afraid that the fix for this is still pending :cry:

**Reply by benjaminh83** [reply]
Hmm. @stuberman mentioned it worked for him now...

**Reply by stuberman** [reply]
I upgraded to v1.17.2-rc2 24 hours ago (at release) and here are my stats (based on a new snapshot of 85G) running coldstore autoprune/universal

`du -sch /.lotus/datastore/* | sort -rh`

> 255G	total
> 156G	/.lotus/datastore/splitstore
> 100G	/.lotus/datastore/chain

**Reply by TippyFlitsUK** [reply]
Thanks guys!!

**Comment by namoo1** [comment]
# Meest NAM

- [X] - If the new lightweight snapshot downloads correctly
- [X] - If the new lightweight snapshot imports correctly

| SplitStore Mode | Directory | Size in GiBs immediately following daemon start |
| ------------------ | ---------- | ---------------------- |
| AutoPrune | /.lotus/datastore/chain | 5G
| AutoPrune | /.lotus/datastore/splitstore | 5.4G 

- [X] If lotus daemon stays in sync
- [X] if you can see the following new symlink path - /.lotus/datastore/splitstore/hot.badger 🔥 🦡
- [X] if your /.lotus/datastore/chain folder is being successfully garbage collected
- [X] if normal mining tasks appear to be processed as expected
- [X] if normal daemon tasks appear to be processed as expected

| SplitStore Mode | Directory | Week 1 | Week 2 | Week 3 | Week 4 |
| ---------------------- | ------------ | -------| --------- | -------- | -------- |
| AutoPrune Mode | /.lotus/datastore/splitstore | 7.0G | 6.5G | 7.0G
| AutoPrune Mode | /.lotus/datastore/chain | 9.5G | 12.0G | 19.0G

After performing "Split store", wdpost failed, but let's see if there are any other causes
```
Daemon:  1.17.1-rc3+calibnet+git.079dab0d7+api1.5.0
Local: lotus-miner version 1.17.1-rc3+calibnet+git.079dab0d7

The capacity of the "splitstore" has decreased...
Is there any other reason?

**Reply by TippyFlitsUK** [reply]
Thanks, Meest!

We haven't had any other reports of this affecting wdpost but please keep us updated! :pray:

**Comment by SBudo** [comment]
- [x] - If the new lightweight snapshot downloads correctly
- [x] - If the new lightweight snapshot imports correctly

| SplitStore Mode  | Directory | Size in GiBs immediately following daemon start |
| ------------- | ------------- | ------------- |
| Discard  | /.lotus/datastore/chain  | 85GiB  |
| Discard  | /.lotus/datastore/splitstore  | 60GiB |

- [x] - If lotus daemon stays in sync
- [x] - if you can see the following new symlink path - /.lotus/datastore/splitstore/hot.badger 🔥 🦡
- [x] - if your /.lotus/datastore/chain folder is being successfully garbage collected
- [x] - if normal mining tasks appear to be processed as expected
- [x] - if normal daemon tasks appear to be processed as expected

| SplitStore Mode  | Directory | Week 1 | Week 2 | Week 3 | Week 4 |
| ------------- | ------------- | ------------- | ------------- | ------------- | ------------- |
| Discard  | /.lotus/datastore/chain  | 85GiB  | 85GiB  | 85GiB  | 85GiB  |
| Discard  | /.lotus/datastore/splitstore  | 129GiB | 148GiB  | 155GiB  | 287GiB  |

**Reply by SBudo** [reply]
Note: in my case, the /.lotus/datastore/splitstore/hot.badger is not a symlink, but just a folder

**Reply by TippyFlitsUK** [reply]
Thanks, Stephane!! You should find this turns into a symlink following the first compaction. :pray:

**Reply by SBudo** [reply]
Correct you are! :-)

**Reply by SBudo** [reply]
Is it expected for the splitstore folder to continue to grow even though I'm using the "discard" mode?

**Reply by TippyFlitsUK** [reply]
There will be some small increases over time Stephane but it should be fairly minimal.

**Comment by cryptowhizzard** [comment]
SplitStore Mode	Directory	Size in GiBs immediately following daemon start
Autoprune	/.lotus/datastore/chain	94GiB
Autoprune	/.lotus/datastore/splitstore	20GiB
* [x]  - If lotus daemon stays in sync
* [x]  - if you can see the following new symlink path - /.lotus/datastore/splitstore/hot.badger 🔥 🦡
* [x]  - if your /.lotus/datastore/chain folder is being successfully garbage collected
* [x]  - if normal mining tasks appear to be processed as expected
* [x]  - if normal daemon tasks appear to be processed as expected

SplitStore Mode	Directory	Week 1	Week 2	Week 3	Week 4
Autoprune	/.lotus/datastore/chain	94GiB			
Autoprune	/.lotus/datastore/splitstore	20GiB

Edit: There is not a symlink for hot.badger. It is a normal directory?
<img width="422" alt="Schermafbeelding 2022-08-31 om 21 49 15" src="https://user-images.githubusercontent.com/36404435/187769293-ab1c05d6-1faa-45cc-a7d4-eb576b9b457f.png">

**Comment by dgreuel2002** [comment]
## Step 5: Reply card
Please provide the information below after completion of the above steps and immediately following the successful start-up of your SplitStore enabled daemon:

- [x] - If the new lightweight snapshot downloads correctly
- [x] - If the new lightweight snapshot imports correctly

| SplitStore Mode | Directory | Size in GiBs immediately following daemon start |
| --- | --- | --- |
| AutoPrune  | /.lotus/datastore/chain | 88G after sync |
| Auto Prune | /.lotus/datastore/splitstore | 44G after sync  |

</br>Please complete the following after observing usual operations:
- [x] - If `lotus daemon` stays in sync
- [x] - if you can see the following new symlink path - `/.lotus/datastore/splitstore/hot.badger` :fire: :badger: 
- [ ] - if your `/.lotus/datastore/chain` folder is being successfully garbage collected
- [x] - if normal mining tasks appear to be processed as expected
- [x] - if normal daemon tasks appear to be processed as expected

We will be asking for updated sizes of your `/.lotus/datastore/chain` and `/.lotus/datastore/splitstore` folders on a weekly basis for a period of 4 weeks. We will publish weekly reminders in `#spx-current-class` but this is obviously dependent on your testing start date. Sizes should be recorded at the end of each week and in `GiB` format. Please consider setting weekly reminders so metrics remain consistent and accurate. :pray:
 
| SplitStore Mode | Directory | Week 1 | Week 2 | Week 3 | Week 4 |
| --- | --- | --- | --- | --- | --- |
| Autoprune | /.lotus/datastore/splitstore | 277G  |  |  |  |
| Autoprune | /.lotus/datastore/chain | 106G |  |  |  |
